tls/s2n_client_hello.c

```diff
@@ -326,16 +354,7 @@ int s2n_client_hello_send(struct s2n_connection *conn)
     GUARD(s2n_stuffer_write_bytes(out, client_protocol_version, S2N_TLS_PROTOCOL_VERSION_LEN));
     GUARD(s2n_stuffer_copy(&client_random, out, S2N_TLS_RANDOM_DATA_LEN));
 
-    /* Generate client session id when empty so that when server sends
-     * an empty session id it is because it doesn't support session resumption
-     */
-    if (conn->session_id_len == 0 && conn->config->use_tickets) {
-        struct s2n_blob session_id = {0};
-        GUARD(s2n_blob_init(&session_id, conn->session_id, S2N_TLS_SESSION_ID_MAX_LEN));
-        GUARD_AS_POSIX(s2n_get_public_random_data(&session_id));
-        conn->session_id_len = S2N_TLS_SESSION_ID_MAX_LEN;
-    }
-
+    GUARD_AS_POSIX(s2n_generate_client_session_id(conn));
```

Comment:
Should this be `GUARD_AS_RESULT`?

Reply:
It's in a function that returns `int` and the function being called returns a `s2n_result`. `GUARD_AS_RESULT` would convert an int into a result.